devonfw/IDEasy#991: Introduce dependencies.json for gradle

hj-lorenz opened the pull request:
Addresses: devonfw/IDEasy#991
Added new file dependencies.json for gradle

jan-vcapgemini reviewed (approved):
LGTM. Please adjust the commit title to `devonfw/DEasy` using git amend and a force push.

hj-lorenz commented:
> LGTM. Please adjust the commit title to `devonfw/DEasy` using git amend and a force push.

Following your suggestion I adjusted the commit title and forced pushed  it. 
Additionally I rebased the branch onto the actual (todays) upstream/master and forced pushed it again.

hohwille reviewed (approved):
@hj-lorenz thanks for the PR. All looks perfect to me. You properly translated the web page with compatibility information to our JSON format. Well done 👍 
We will wait for the local test of this change with current IDEasy product (not including the according code changes related this this). When this still works without breaking installation of gradle, we can merge. Otherwise we must discuss with our users and create a migration plan.

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[7.3,7.5)": [
+    {
+      "tool": "java",
+      "versionRange": "[17,)"
+    }
+  ],
+  "[7.5,7.6)": [
+    {
+      "tool": "java",
+      "versionRange": "[18,)"
+    }
+  ],
+  "[7.6,8.3)": [
+    {
+      "tool": "java",
+      "versionRange": "[19,)"
+    }
+  ],
+  "[8.3,8.5)": [
+    {
+      "tool": "java",
+      "versionRange": "[20,)"
+    }
+  ],
+  "[8.5,8.8)": [
+    {
+      "tool": "java",
+      "versionRange": "[21,)"
+    }
+  ],
+  "[8.8,8.10)": [
+    {
+      "tool": "java",
+      "versionRange": "[22,)"
+    }
+  ],
+  "[8.10,)": [
+    {
+      "tool": "java",
+      "versionRange": "[23,)"
```
Thanks for the link to the source.
However, this is not the minimum but the maximum JDK version supported by Gradle.
Hence, your PR needs to be updated.

hj-lorenz replied:
Yes, you are right. 
Following your suggestion making actual minimum to max-version and setting minimum "8" for all dependencies I adjusted the PR.. 
Please check again.

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[5.4,6.0)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,12]"
```
```suggestion
      "versionRange": "[8,13)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[6.0,6.3)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,13]"
```
```suggestion
      "versionRange": "[8,14)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[6.3,6.7)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,14]"
```
```suggestion
      "versionRange": "[8,15)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[6.7,7.0)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,15]"
```
```suggestion
      "versionRange": "[8,16)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[7.0,7.3)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,16]"
```
```suggestion
      "versionRange": "[8,17)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[7.3,7.5)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,17]"
```
```suggestion
      "versionRange": "[8,18)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[7.5,7.6)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,18]"
```
```suggestion
      "versionRange": "[8,19)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[7.6,8.3)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,19]"
```
```suggestion
      "versionRange": "[8,20)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[8.3,8.5)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,20]"
```
```suggestion
      "versionRange": "[8,21)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[8.5,8.8)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,21]"
```
```suggestion
      "versionRange": "[8,22)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[8.8,8.10)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,22]"
```
```suggestion
      "versionRange": "[8,23)"
```

hohwille commented on gradle/gradle/dependencies.json:
```diff
@@ -0,0 +1,98 @@
+  "[8.10,)": [
+    {
+      "tool": "java",
+      "versionRange": "[8,23]"
```
```suggestion
      "versionRange": "[8,24)"
```

hohwille reviewed (commented):
@hj-lorenz thanks for the update. You got it almost.
However, the version seems to be the supported version so it is not directly the max value.
It gets most obvious here:
```
 "(,4.3)": [
    {
      "tool": "java",
      "versionRange": "[8,8]"
    }
  ],
```

There is no Java version at all that will match `[8,8]`. It could only be matched by a version named exactly `8` but there exists no such version.
We need to define a supreme for max increased by one.
